README
NAME
    Test::Mini - Clean Unit Testing

SYNOPSIS
    Real Documentation is coming. In the meantime, enjoy the montage!

  Traditional Style
      package TraditionalTest;

      use Test::Mini 'sugarfree';
      use Test::Mini::Assertions;

      sub setup    { 'This runs before each test...' }
      sub teardown { 'This runs after each test...' }

      sub test_assert { assert 1, 'I should pass' }
      sub test_refute { refute 0, 'I should fail' }
      sub test_skip   { skip "I've got better things to do" }

      1;

  Sweetened Style
      use Test::Mini;

      testcase SugaryTest {
        setup    { 'This runs before each test...' }
        teardown { 'This runs after each test...' }

        test passes() { assert 1, 'I should pass' }
        test refute() { refute 0, 'I should fail' }
        test skip()   { skip "I've got better things to do" }
      }

AUTHOR
    Pieter van de Bruggen, <pvande@cpan.org<gt>

COPYRIGHT AND LICENSE
    Copyright (C) 2010 by Pieter van de Bruggen

    This library is free software; you can redistribute it and/or modify it
    under the same terms as Perl itself, either Perl version 5.8.9 or, at
    your option, any later version of Perl 5 you may have available.



Changes
Revision history for Test-Mini

0.01    Date/time
        First version, released on an unsuspecting world.



lib/Test/Mini/Unit.pm
use MooseX::Declare 0.31;

use 5.008;

class Test::Mini::Unit
{
  use aliased 'MooseX::Declare::Syntax::Keyword::Class',     'ClassKeyword';
  use aliased 'MooseX::Declare::Syntax::Keyword::Role',      'RoleKeyword';
  use aliased 'Test::Mini::Unit::Syntax::Keyword::TestCase', 'TestCaseKeyword';

  method import(ClassName $class: %args)
  {
    my $caller = caller();

    strict->import;
    warnings->import;

    for my $keyword (
      ClassKeyword->new(identifier => 'class'),
      RoleKeyword->new(identifier => 'role'),
      TestCaseKeyword->new(identifier => 'testcase'),
    ) {
      $keyword->setup_for($caller, %args, provided_by => $class);
    }
  }

  use Test::Mini::Unit::Runner;

  END {
    $| = 1;
    return if $?;
    $? = Test::Mini::Unit::Runner->new_with_options()->run();
  }
}

__END__
# Below is stub documentation for your module. You'd better edit it!

=head1 NAME

Test::Mini::Unit - Clean Unit Testing with Moose

=head1 VERSION

version 0.050200

=head1 SYNOPSIS

Real Documentation is coming.  In the meantime, enjoy the montage!

=head2 Traditional Style

  package TraditionalTest;

  use Test::Mini::Unit;
  use Test::Mini::Unit::Assertions;

  sub setup    { 'This runs before each test...' }
  sub teardown { 'This runs after each test...' }

  sub test_assert { assert 1, 'I should pass' }
  sub test_refute { refute 0, 'I should fail' }
  sub test_skip   { skip "I've got better things to do" }

  1;

=head2 Classical Style

  use Test::Mini::Unit;

  class ClassicalTest extends Test::Mini::Unit::TestCase
  {
    use Test::Mini::Unit::Assertions;

    method setup()    { 'This runs before each test...' }
    method teardown() { 'This runs after each test...' }

    method test_assert() { assert 1, 'I should pass' }
    method test_refute() { refute 0, 'I should fail' }
    method test_skip()   { skip "I've got better things to do" }
  }

=head2 Sweetened Style

  use Test::Mini::Unit;

  testcase SugaryTest
  {
    setup    { 'This runs before each test...' }
    teardown { 'This runs after each test...' }

    test passes() { assert 1, 'I should pass' }
    test refute() { refute 0, 'I should fail' }
    test skip()   { skip "I've got better things to do" }
  }

=head1 AUTHOR

Pieter Vande Bruggen, E<lt>pvande@cpan.org<gt>

=head1 COPYRIGHT AND LICENSE

Copyright (C) 2010 by Pieter Vande Bruggen

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself, either Perl version 5.8.9 or,
at your option, any later version of Perl 5 you may have available.

=cut

Makefile.PL
use strict;
use warnings;

BEGIN { require 5.008; } 

use ExtUtils::MakeMaker 6.31;
use ExtUtils::Manifest  1.56;

my $SKIP_FILES = <<SKIP;
SKIP

my %WriteMakefileArgs = (
    'NAME'     => 'Test::Mini',
    'ABSTRACT' => 'Lightweight xUnit Testing for Perl',
    'VERSION'  => 'v0.6.0_5',
    'AUTHOR'   => 'Pieter van de Bruggen <pvande@cpan.org>',
    'LICENSE'  => 'perl',
    'PREREQ_PM' => {
        'Try::Tiny'        => '0.04',
        'Scalar::Util'     => '1.21',
        'List::Util'       => '1.21',
        'List::MoreUtils'  => '0.22',
        'MRO::Compat'      => '0.10',
        'Getopt::Long'     => '2.38',
        'Time::HiRes'      => '1.9719',
        'Exception::Class' => '1.30',
        'namespace::clean' => '0.14',
        'Data::Inspect'    => '0.03',
        'Text::Outdent'    => '0.01',
    },
    'BUILD_REQUIRES' => {
        'Test::More' => '0.96',
        'B'          => '0',
        'aliased'    => '0.30',
        'IO::Scalar' => '2.110',
    },
    'CONFIGURE_REQUIRES' => {
         'ExtUtils::MakeMaker' => '6.31',
         'ExtUtils::Manifest'  => '1.56',
    },
#   'DIST_REQUIRES' => {
#       'ShipIt'                 => '0.55',
#       'ShipIt::Step::Manifest' => '0.1',
#   },
    'test' => {
        'TESTS' => 't/*.t t/*/*.t t/*/*/*.t t/*/*/*/*.t t/*/*/*/*/*.t',
    },
    'dist' => {
        'PREOP'  => '',
        'POSTOP' => 'make realclean; rm MANIFEST',
    },
);

delete $WriteMakefileArgs{BUILD_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.56) };

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

(my $DISTNAME = $WriteMakefileArgs{NAME}) =~ s/::/-/g;

# Create a MANIFEST.SKIP file.
open(GITIGNORE, '<.gitignore');
open(SKIP, '>MANIFEST.SKIP');
print SKIP "#!include_default\n";
print SKIP "MANIFEST.SKIP.*\n";
print SKIP "^$DISTNAME.*\n";
print SKIP <GITIGNORE>;
print SKIP $SKIP_FILES;
close SKIP;
close GITIGNORE;

# Create a new MANIFEST.
ExtUtils::Manifest::mkmanifest();

# Write the new Makefile.
WriteMakefile(%WriteMakefileArgs);

# Cleanup the MANIFEST.SKIP files.
unlink glob 'MANIFEST.{,SKIP}*';
